src/config/reader.rs

```diff
@@ -57,6 +57,44 @@ impl ConfigReader {
             return Ok(config_module);
         }
 
+        let protobuf_links = links
+            .iter()
+            .filter(|v| v.type_of == LinkType::Protobuf)
+            .collect::<Vec<_>>();
+        let protobuf_links_paths = protobuf_links
+            .iter()
+            .map(|v| Self::resolve_path(&v.src, parent_dir))
+            .collect::<Vec<String>>();
```

Comment:
Don't convert to iter and then to vec and then back to iter. Perform all the operations while we have an iter.